src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ManageContent/KeyStatisticService.cs

```diff
@@ -165,46 +165,14 @@ await _dataBlockService.IsUnattachedDataBlock(releaseId, dataBlock)
             .OnSuccess(_userService.CheckCanUpdateRelease)
             .OnSuccess(async release =>
                 await _persistenceHelper.CheckEntityExists<KeyStatistic>(keyStatisticId, query =>
-                    query
-                        .Include(keyStat => (keyStat as KeyStatisticDataBlock)!.DataBlock) // TODO Include can be removed in EES-3988
-                        .Where(keyStat => keyStat.ReleaseId == release.Id)))
-            // TODO: Can remove the below OnSuccess in EES-3988
-            // Key stats were previously associated with a KeyStatistic content section,
-            // which meant their data blocks had a ContentSectionId set. The below code
-            // ensures ContentSectionId is unset when a key stat is removed. If it isn't
-            // unset, that data block won't become available for selection for content again.
-            .OnSuccess(async keyStat =>
-            {
-                if (keyStat is KeyStatisticDataBlock keyStatDataBlock)
-                {
-                    var dataBlock = keyStatDataBlock.DataBlock;
-                    dataBlock.ContentSection = null;
-                    dataBlock.ContentSectionId = null;
-                    _context.ContentBlocks.Update(dataBlock);
-                    await _context.SaveChangesAsync();
-                }
-                return keyStat;
-            })
+                    query.Where(keyStat => keyStat.ReleaseId == release.Id)))
```

Comment:
I don't think this is going to have any effect because the variable named `query` here is the hydration function, but I might be wrong.

If you write a unit test for this where the key statistic isn't for the release I reckon it would pass currently.

I suggest this is rewritten

```
            .OnSuccessDo(_userService.CheckCanUpdateRelease)
            .OnSuccess(() => _persistenceHelper.CheckEntityExists<KeyStatistic>(q => q
                .Where(ks => ks.Id == keyStatisticId && ks.ReleaseId == releaseId)))
```